mantidproject/mantid
# Sync instrument across components in the reflectometry GUI

This PR adds functionality to synchronise the instrument across all Batch tabs in the ISIS Reflectometry GUI:
- Add notifications to the MainWindowPresenter when a request to change the instrument is made.
- Move loading/storing the instrument from the BatchPresenter to the MainWindowPresenter.
- Also add notifications to allow updating the instrument without changing the name (this is required when restoring defaults because we want to re-load the instrument).
- Make a minor improvement to the RunsPresenter to only reset the search results when the instrument has been successfully changed.
- Disable changing the Instrument if processing or autoprocessing is running.
- Rename notification functions to be more clear and consistent.
- Remove some unused code regarding the SlitCalculator.

**To test:**

Should be tested by someone at ISIS.

- Open the ISIS Reflectometry interface
- Check that the instrument is the same in the top left and bottom right drop-down boxes on all Batch tabs.
- Change the instrument and check it updates all tabs.
- Try adding/removing batch tabs and check the instrument is the same as the other tabs.
- Try changing some settings on the Experiment and Instrument tabs and then hit Restore Defaults and check they revert to original values. They should also revert to the relevant defaults when you change the instrument.
- On the Runs tab, enter investigation ID `1120015` and click Search. Log in to ICat if requested. Now change the instrument and check that the search ID and results are cleared.
- Click Search again and select all runs and click Transfer. Then click Process to process all runs. Check that the Instrument drop-down is disabled on all Batch tabs (including new ones).
- Click Pause and check that the drop-down is re-enabled on all tabs when processing is finished.
- Click Autoprocess and check the same behaviour. When all runs finish processing the drop-down should still remain disabled (because autoprocessing is polling for new runs) - it should only be re-enabled when you Pause autoprocessing.

Fixes #26536 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
